config.toml.example

```diff
@@ -292,6 +292,10 @@ changelog-seen = 2
 # on this runtime, such as `-C profile-generate` or `-C instrument-coverage`).
 #profiler = false
 
+# Use the optimized LLVM C intrinsics for `compiler_builtins`, rather than Rust intrinsics.
+# Requires the LLVM submodule to be managed by bootstrap (i.e. not external).
+#optimized-compiler-builtins = false
```

Comment:
We should default to `true` IMO, and have people set to `false` if they have an unsupported configuration (and thus get an error message) or if they need to disable it for other reasons.

Setting this to `false` generates a broken compiler when using GCC < 9.4 as the linker on `aarch64-unknown-linux-gnu`, and there might be other targets that produce broken compilers with this flag off. I wouldn't want that to be the default, especially because it's super hard to debug those failures.

There should also be a comment explaining the downsides of setting it to `false`, and explaining what to do if LLVM is not managed by `bootstrap` to set it to `true` (see my other comment).

Reply:
If there were an easy way to tell when this breaks, I would have done that instead of adding the new configuration option. The original problem in https://github.com/rust-lang/rust/issues/101172 was that the cc crate doesn't support cross compiling to Windows; there may be other cases where it's broken I don't know about, and it's very hard to tell that it's related to this option.

Given that I would prefer to keep this defaulting to false. I'm happy to add the documentation you suggested.

Reply:
My concern is that there is no good default for this flag. Setting it to `true` could cause [cross-compilation errors](https://github.com/rust-lang/rust/issues/101172) or [wrong architecture code to be included when cross-compiling](https://github.com/rust-lang/rust/issues/105065), while setting it to `false` results in the compiler being unable to produce any binary on a Tier 1 platform.

The problems when the setting is configured to `false` are IMO equally hard to diagnose as the problems when the setting is configured to `true`, I was only managed to figure out the missing flag was the cause by doing a PR-by-PR compile+bisect.